### PR TITLE
Feat#1685: Unit test cases for User groups SPA

### PR DESCRIPTION
**Pull request by ghanlohar**
# Closes/Fixes/Resolves

### 
ONEPLAT-1685
# Explain the feature/fix

- Unit test cases for User groups SPA

## Does this PR introduce a breaking change

No
## Screenshots

<details>
<summary>View Screenshots</summary>

Screenshot 2021-12-27 at 12 43 22 PM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/user-group-spa/src/components/Form.jsx [hunk 1]
```diff
@@ -126,7 +126,8 @@ function GroupForm () {
 
   function addGroup ( group ) {
     GroupsAPI.addGroup(group)
-      .then((res) => {
+      .then( ( res ) => {
+       console.log("Yay.. ", res);
```
Comment (riginoommen) [inline packages/user-group-spa/src/components/Form.jsx:130]:
```suggestion
```

**Review by riginoommen — CHANGES_REQUESTED** [review]
Just ned to remove a console.log

File: packages/user-group-spa/src/components/Form.jsx (hunk 1 above)
Comment (ghanlohar) [reply packages/user-group-spa/src/components/Form.jsx]: Done

**Review by riginoommen — APPROVED** [review]

File: packages/user-group-spa/src/components/Group.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        ctx.status(200),
+        ctx.json(groupData)
+      );
+    })
+  );
+  // Mocking and returning mocked value from useParam() of react-router-dom
+  jest.spyOn(Router, "useParams").mockReturnValue({ cn: "one-platform" });
+
+  const { getByText, container } = render(
+    <MemoryRouter>
```
Comment (akhilmhdh) [inline packages/user-group-spa/src/components/Group.test.js:29]: Is there any reason we are wrapping every component with the provider and memory router

Comment (ghanlohar) [reply packages/user-group-spa/src/components/Group.test.js:29]: @akhilmhdh Every component updates breadcrumb(in `useEffect()`) so the provider wrapping is needed. And to provide the router context it's recommended to wrap unit tests in one of the Router components hence wrapping`MemoryRouter`.

**Review by akhilmhdh — APPROVED** [review]
LGTM

**Pull request merged**